File: 0268-missing-number/0268-missing-number.py
from typing import List

class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        result = 0

        # XOR all indices from 0 to len(nums)
        for i in range(len(nums) + 1):
            result ^= i

        # XOR all numbers in the array
        for num in nums:
            result ^= num

        return result

File: 0268-missing-number/0268-missing-number.ts
function missingNumber(nums: number[]): number {
    let result = 0;

    // XOR all indices from 0 to nums.length
    for (let i = 0; i <= nums.length; i++) {
        result ^= i;
    }

    // XOR all numbers in the array
    for (const num of nums) {
        result ^= num;
    }

    return result;
}
